### ISSUE TITLE
Replace Location string field with country flags/dropdown selection

### ISSUE THREAD

**Issue by Noxxaa** [TASVideos/tasvideos#1997]
The current system of having a string field for location hasn't been in vogue since the 2000s when this first became a regular thing on forums. It also opens up room for abuse, putting profanity in there, or people not exactly understanding the limits of what information they should share on the internet and doxxing their home address (this is, concerningly, a semi-regular occurrence that moderators have to deal with). This borders between being an annoyance to moderation staff, to becoming a legal liability in the worst case.

I propose replacing the Location string field with a dropdown Country flag/field, in the same way SRCom does it. This would be visible on the user's profile like this:

![image](https://github.com/user-attachments/assets/7c08a8f0-f1db-43e0-904a-33e88e2f491d)

This would be set in the user settings in a dropdown like this:

![image](https://github.com/user-attachments/assets/44e342ae-649e-44b0-b6e3-e6869c5325fd)

And optionally a sub-region field for relevant countries (e.g. U.S. states):

![image](https://github.com/user-attachments/assets/20c9db11-a6a5-463f-bb2c-72517e7d3885)

The country flag is both more colorful and compact than the current location field, and makes users easier to distinguish and identify. It also does not give users an outlet to potentially doxx themselves to any significant extent.

**Comment by YoshiRulz** [comment]
Perhaps these users are misled by the location of the field: info shown publically on your profile and posts is intermingled with settings which affect the behaviour of the frontend, which are intuitively private. They would ideally be on separate pages.

**Comment by YoshiRulz** [comment]
There's no "only visible on mobile", it's entirely dependent on whether the font that your browser ends up using includes (coloured) glyphs for those characters / combining sequences.
Because Chrome on Windows didn't have a font covering those glyphs, Masterjun set up a web font in #2183.

**Comment by Masterjun3** [comment]
Yes it is possible, but we're currently relying on a separate font package provided by some dude who spent a lot of effort getting the details right. Specifically for the country flags.
Setting up all the stuff to include special other emojis is possible, but it's honestly more effort than I'm personally willing to put in. We'd then also have to self-host the font, which is unfortunate.

**Comment by YoshiRulz** [comment]
The EU flag already works.
Any combining sequence apart from country flags, such as the pirate flag, won't work because those codepoints [aren't included in the font file](https://github.com/talkjs/country-flag-emoji-polyfill/blob/8ad89980d0ffea4432aae16c54ce58714f13ad3b/build/make-font.sh#L19-L26). I imagine if the original Twemoji font was referenced then they would.

While investigating this, I found out Bootstrap already includes an emoji font ([Noto Color Emoji](https://fonts.google.com/noto/specimen/Noto+Color+Emoji)), which covers all the flags. But applying that to an element causes regular text to be spaced incorrectly (was the `@font-face` set up wrong?):
<img width="281" height="38" alt="Image" src="https://github.com/user-attachments/assets/1a00c754-656f-4d0e-9af5-b881bdef90d5" />
What *does* work is `font-family: initial;`, but that might be the system font again.
